action.yml

```diff
@@ -0,0 +1,58 @@
+  
+  In 2021, Github released the createCommitOnBranch GraphQL mutation, which makes it easier to add, update, and delete files in a branch of a repository. This new API offers a simpler way to commit changes compared to the existing Git database REST APIs. With the new createCommitOnBranch mutation, you do not need to manually create blobs and trees via separate API calls before creating the commit. This allows you to add, update, or delete multiple files in a single API call.
+  
+  The push-signed-commits composite action uses the new createCommitOnBranch GraphQL endpoint to create verified commits on a remote branch. This GraphQL API extracts authorship information from the credential used for authentication, and automatically marks commits created using Github App installation credentials as "verified". 
+  
+  Read more about this new mutation and its conveniences here: https://github.blog/changelog/2021-09-13-a-simpler-api-for-authoring-commits/
+
+inputs:
+  github-token:
+    description: 'GitHub token'
+    required: true
+  owner:
+    description: 'Repository owner'
+    required: true
+    default: ${{ github.repository_owner }}
+  repo:
+    description: 'Repository name'
+    required: true
+    default: ${{ github.repository }}
+  local_branch_name:
+    description: 'Local branch name'
+    required: true
+  remote_branch_name:
+    description: 'Remote branch name - should not include refs/heads/ prefix or origin/ prefix'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Mask token
+      run: echo "::add-mask::${{ inputs.github-token }}"
+      shell: bash
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install requests
+      shell: bash
+    - name: Run script
+      run: |
+        python ${{ github.action_path }}/create_commits.py ${{ inputs.github-token }} ${{ inputs.repo }} ${{ inputs.local_branch_name }} ${{ inputs.remote_branch_name }}
+      shell: bash
+    # The createCommitOnBranch endpoint will dynamically generate authorship and commit verification information, which will result in different commit hashes on the remote vs. local branch. 
+    # If the creation of remote commits is successful, a final step runs. This step will reset the local branch to have the same authorship info, verification status, and therefore commit hash (which is deterministically derived from commit contents) as the remote. 
+    - name: Reset the local branch to the remote branch
+      if: success()
+      run: |
+        git fetch origin
```

Comment:
i recommend making this specific (and you can specify multiple refs after the remote) - instead of fetching _all_ refs from the remote (which can be very slow)

Reply:
it would be nice to parameterize the remote name, cuz there are use-cases for this action where there are multiple remotes (e.g. translation syncing)

Reply:
omg yes great minds. I was doing this just as your review came in